app/models/mixins/event_mixin.rb

```diff
@@ -36,10 +36,48 @@ def events_table_name(assoc)
     events_assoc_class(assoc).table_name
   end
 
+  #TODO: Remove me, this is a hack so the UI always calls what I want to call
+  # The UI should call ems_event_filter or miq_event_filter for management events and policy events
+  def event_stream_filter
+    miq_event_filter
+  end
+
+  def ems_event_filter
+    { self.class.ems_event_filter_column => id }
+  end
+
+  def miq_event_filter
+    filter = { self.class.miq_event_filter_column => id }
+    filter["target_type"] = self.class.base_class.name if filter.key?("target_id")
+    filter
```

Comment:
This is a hack but if we're filtering on the `target_id`, we need to specify the `target_type` also.  Note, `target` is the only `polymorphic` association.